### ISSUE TITLE
Bug: (fill in) eslint cli is not work well 

### ISSUE THREAD

**Issue by vaynevayne** [eslint/eslint#18572]
### Environment

Node version: 
npm version: 
Local ESLint version: 
Global ESLint version: 
Operating System: 

### What parser are you using?

@typescript-eslint/parser

### What did you do?

run 
`npm lint`

### What did you expect to happen?

get error  in App.tsx  `Parameter 'arg' implicitly has an 'any' type.(7006)`

### What actually happened?

nothing happend

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/vitejs-vite-obdkyw?file=src%2FApp.tsx

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
This error is from `tsc` not ESLint.

**Comment by vaynevayne** [comment]
hi @nzakas What I mean is, I need eslint to report this error. The fundamental reason is that eslint reports all issues with tsc. My usage method is to perform eslint checks during commit, and tsc checks during online build. The problem I encountered was that commit multiple times and push, but in the end, tsc checks did not pass during build, which is a very painful thing. Excuse me, do I need to use both eslint and tsc simultaneously? When I built some projects, I didn't use tsc and only used eslint. If eslint and tsc are not consistent, then I must add tsc
